File: cpp/260_SingleNumberIII.cpp
class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        int diff = 0;
        for(auto num : nums){
            diff ^= num;
        }
        diff &= -diff;
        
        vector<int> ret(2,0);
        for(auto num : nums){
            if((num & diff) == 0) ret[0] ^= num;
            else ret[1] ^= num;
        }
        return ret;
    }
};

File: js/260_SingleNumberIII.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var singleNumber = function(nums) {
    var diff = 0;
    for(var i in nums){
        diff ^= nums[i];
    }
    diff &= -diff;
    var ret = [0,0];
    for(var j in nums){
        if((nums[j] & diff) === 0) ret[0] ^= nums[j];
        else ret[1] ^= nums[j];
    }
    return ret;
};